pkg/authorization/rewrite/middleware_test.go

```diff
@@ -0,0 +1,142 @@
+package rewrite_test
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"reflect"
+	"testing"
+
+	"github.com/brancz/kube-rbac-proxy/pkg/authorization/rewrite"
+)
+
+func TestRewriteParamsMiddleware(t *testing.T) {
```

Comment:
did we have any unit tests before the refactoring?

Reply:
Yes, we had. I copied them into the new unit tests and added some on top of it.
I needed to split the unit tests into the three different components, which where just one huge "integration" tests for the package.
I wanted to have a similar integration test, but the complexity of all the mocks didn't make it easily readable.